src/Commands/SiteStudioSettings.php

```diff
@@ -0,0 +1,225 @@
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Entity Type manager.
+   * @param \Drupal\Component\Uuid\UuidInterface $uuid
+   *   A service for generating UUIDs.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, UuidInterface $uuid) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->uuid = $uuid;
+  }
+
+  /**
+   * Generate all the Drupal content types from DEG sheet.
+   *
+   * @command deg:generate:site-studio
+   * @aliases deg:ss $type
+   * @usage drush deg:ss color
+   *   Generates site studio fonts if not present.
+   * @usage drush deg:ss --update
+   *   Generate site studio fonts if not present also updates existing.
+   * @option update Updates existing entity types and creates new if not present.
+   */
+  public function generateSiteStudioSettings($options = ['update' => FALSE, 'type' => NULL]) {
+    $this->io()->success('Generating site studio settings.');
+    $this->updateMode = $options['update'];
+    $mode = 'create';
+    if ($this->updateMode) {
+      $mode = 'update';
+    }
+    $data = $this->getDataFromSheet(DstegConstants::SITE_STUDIO_SETTINGS);
+    // Site studio settings generated from DST sheet.
+    $settings = $this->getSiteStudioData($data, $options['type']);
+    foreach ($settings as $key => $value) {
+      $type = $value['type'];
+      $setting = $value['entity']['id'];
+      $entity_type = ($type === 'Font') ? 'cohesion_font_stack' : 'cohesion_color';
+      $entity = $this->entityTypeManager->getStorage($entity_type)->load($setting);
+      if (!empty($entity)) {
+        if ($this->updateMode && $data[$key][$this->implementationFlagColumn] === $this->updateFlag) {
+          $this->updateEntityType($entity, $value['entity']);
+          $this->io()->success($type . ' ' . $setting .  " updated.");
+          continue;
+        }
+        $this->io()->warning($type . ' ' . $setting . " Already exists. Skipping creation.");
+        continue;
+      }
+      $status = $this->entityTypeManager->getStorage($entity_type)->create($value['entity'])->save();
+      if ($status === SAVED_NEW) {
+        $this->io()->success($type . ' ' . $setting .  " is successfully created.");
+      }
+    }
+  }
+
+  /**
+   * Get data needed for site studio entity.
+   *
+   * @param array $data
+   *   Array of Data.
+   *
+   * @return array|null
+   *   Font stack data.
+   */
+  private function getSiteStudioData(array $data, $param) {
+    $site_studio_data = [];
+    foreach ($data as $item) {
+      // To allow existing entity update.
+      if (!$this->validateMachineName($item['machine_name'])) {
+        continue;
+      }
+      // Type value will distinguish between the cohesion entities.
+      // Command with param -> type = param.
+      // Command with no param -> type value from sheet data to be considered.
+      $type = $param ?: strtolower($item['type']);
+      switch ($type) {
```

Comment:
**Avoid code duplicacy in switch cases**
Avoid duplicate code statements in switch case.

Reply:
Each switch case makes a call to different methods based on cohesion types -> font stack, font library, color.